lib/Service/AttachmentService.php

```diff
@@ -215,79 +203,89 @@ private function getMediaFilePreviewFile(string $mediaFileName, File $textFile):
-	 * @throws \OCP\Files\InvalidPathException
-	 * @throws \OC\User\NoUserException
 	 */
-	public function getMediaFileMetadataPrivate(int $documentId, string $mediaFileName, string $userId): ?array {
-		$textFile = $this->getTextFile($documentId, $userId);
-		return $this->getMediaFileMetadata($mediaFileName, $textFile);
-	}
+	public function getAttachmentList(int $documentId, ?string $userId = null, ?Session $session = null, ?string $shareToken = null): array {
+		if ($shareToken) {
+			$textFile = $this->getTextFilePublic($documentId, $shareToken);
+		} elseif ($userId) {
+			$textFile = $this->getTextFile($documentId, $userId);
+		} else {
+			throw new NotPermittedException('Unable to read document');
+		}
 
-	/**
-	 * @param int $documentId
-	 * @param string $mediaFileName
-	 * @param string $shareToken
-	 * @return array|null
-	 * @throws NotFoundException
-	 * @throws NotPermittedException
-	 * @throws \OCP\Files\InvalidPathException
-	 * @throws \OC\User\NoUserException
-	 */
-	public function getMediaFileMetadataPublic(int $documentId, string $mediaFileName, string $shareToken): ?array {
-		$textFile = $this->getTextFilePublic($documentId, $shareToken);
-		return $this->getMediaFileMetadata($mediaFileName, $textFile);
-	}
+		try {
+			$attachmentDir = $this->getAttachmentDirectoryForFile($textFile);
+		} catch (NotFoundException) {
+			return [];
+		}
 
-	/**
-	 * @param string $mediaFileName
-	 * @param File $textFile
-	 * @return array|null
-	 * @throws NotFoundException
-	 * @throws NotPermittedException
-	 * @throws \OCP\Files\InvalidPathException
-	 * @throws \OC\User\NoUserException
-	 */
-	private function getMediaFileMetadata(string $mediaFileName, File $textFile): ?array {
-		$attachmentFolder = $this->getAttachmentDirectoryForFile($textFile, true);
-		$mediaFile = $attachmentFolder->get($mediaFileName);
-		if ($mediaFile instanceof File) {
-			return [
-				'size' => Util::humanFileSize($mediaFile->getSize()),
-				'mtime' => $mediaFile->getMTime(),
+		$shareTokenUrlString = $shareToken
+			? '&shareToken=' . urlencode($shareToken)
+			: '';
+		$urlParamsBase = $session
+			? '?documentId=' . $documentId . '&sessionId=' . $session->getId() . '&sessionToken=' . urlencode($session->getToken()) . $shareTokenUrlString
+			: '?documentId=' . $documentId . $shareTokenUrlString;
+
+		$attachments = [];
+		foreach ($attachmentDir->getDirectoryListing() as $node) {
+			if (!($node instanceof File)) {
+				// Ignore anything but files
+				continue;
+			}
+			$isImage = in_array($node->getMimetype(), AttachmentController::IMAGE_MIME_TYPES, true);
+			$name = $node->getName();
+			$attachments[] = [
+				'fileId' => $node->getId(),
+				'name' => $name,
+				'size' => Util::humanFileSize($node->getSize()),
+				'mimetype' => $node->getMimeType(),
+				'mtime' => $node->getMTime(),
+				'isImage' => $isImage,
+				'shareToken' => $shareToken,
+				'davPath' => '/' . implode('/', array_slice(explode('/', $node->getPath()), 3)),
```

Comment:
Not entirely sure this will always get the right path, maybe rather use getRelativePath from the user folder?
Also to clarify would be what is dav path considered for a public share link

Reply:
Good catch. Changed now to use `userFolder->getRelativePath()` if there's a user and set `davPath` to `null` in public shares. We most likely don't have DAV access to the attachments in public shares anyway.